lib/rubocop/cop/rspec/factory_bot/attribute_defined_statically.rb

```diff
@@ -88,6 +89,20 @@ def autocorrect(node)
 
           private
 
+          def offensive_receiver?(receiver, node)
```

Comment:
I guess we can manage to solve this problem by mixing custom methods with node pattern. Example:

```ruby
          def capture_arg(sym)
            @block_argument = sym
          end

          def from_capture?(node)
            node.node_parts == [@block_argument]
          end
```

and then we need to use the capture arg and the function to compare the nodes:

```ruby
          def_node_matcher :value_matcher, <<-PATTERN
            (send { self nil? (lvar #capture_arg) } !#reserved_method? $...)
          PATTERN

          def_node_search :factory_attributes, <<-PATTERN
            (block (send {nil? #from_capture?} #attribute_defining_method? ...) _ { (begin $...) $(send ...) } )
          PATTERN
```